src/gennylib/src/PoolManager.cpp

```diff
@@ -53,9 +55,22 @@ auto createPool(const std::string& name,
 }  // namespace genny::v1
 
 
-mongocxx::pool::entry genny::v1::PoolManager::client(const std::string& name,
-                                                     size_t instance,
-                                                     const Node& context) {
+mongocxx::pool::entry genny::v1::PoolManager::create_client(const std::string& name,
+                                                            size_t instance,
+                                                            const Node& context) {
+    auto pool_entry = this->_create_client(name, instance, context);
+    bool prewarm = context["Clients"][name]["PreWarm"].maybe<bool>().value_or(true);
+    if (prewarm && !_dryRun) {
+        auto ping = bsoncxx::builder::stream::document{} << "ping" << 1 << bsoncxx::builder::stream::finalize;
```

Comment:
Upon reading this, it is not clear how this solves the issue. It seems to rely upon the particular connection dynamic of Genny.

To improve clarity, it would be useful to add a comment explaining why pre-warming has been implemented and why it is being done here instead of pre-warming the entire pool with a loop upon initial pool construction (which would be the naive assumption). The comment does not need to be here, but wherever it makes sense.

Reply:
I've added more details for these questions in the PR description.